ambry-api/src/main/java/com.github.ambry/config/RouterConfig.java

```diff
@@ -54,6 +54,20 @@
   @Default("2")
   public final int routerScalingUnitMaxConnectionsPerPortSsl;
 
+  /**
+   * The number of connections to local DC created in the startup.
+   */
+  @Config("router.connections.warm.up.count.per.port.to.local.dc")
```

Comment:
I think we should model this as a percentage of the total since that will scale as soon as the total is changed (unless that is not desired). What is the rationale behind having a number?

Reply:
Also, what is "per port" here? plaintext and SSL? Once again, percentage may be better than a number if that is the case

Reply:
I struggled a little while. The main reason is, we don't know what is a good `router.connections.warm.up.max.waiting.time` if percentage is used. 

The code checks port type to decide what type of connection(ssl/plain text) it should connect, so "per port" is used. As only one type of connection is used, one config var is good enough. 
(Same thing if we switch to percentage).

Reply:
Changed code a little and `Percentage` is used.